src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java

```diff
@@ -853,6 +904,13 @@ public ICraftingLink submitJob(final IGrid g, final ICraftingJob job, final Base
             this.playersFollowingCurrentCraft.clear();
 
             if (ci.commit(src)) {
+                for (Runnable onCancelListener : onCancelListeners) {
+                    onCancelListener.run();
```

Comment:
Can you explain why cancel listeners should be triggered at craft startup? My understanding is that if the craft starts successfully, the cluster is empty (free of crafts), so there should be no listeners (since there is no craft to follow). Please correct me if I'm wrong.

Reply:
Well, the initial reason is serialization of non-static inner class (passed by exteral application) is dangerous and can cause inconsistance. But it's caller's responsibility. Maybe it's not necessary to clean those states because it's not the work that AE itself should do, emm.
You can tell me how do you think of that.

Reply:
Sorry, I didn't get your answer. Why are you talking about serialization *here*? I'm asking about the logic: why do you call cancel listeners at the start of new crafting job?

Reply:
![3b49c9c2fbdd14be137b6643f57ad725](https://github.com/user-attachments/assets/89b614db-a767-499b-b37f-7843fa054799)
The code right before also clears the players follows the craft before. 
Maybe it's overconsideration, but I did observed there's inconsistency during the lifecycle of task, such as listeners are still there even if it's canceled.
But I agree, resetting listeners in cancel and complete is probably enough. Let me think of what else I have missed.

Reply:
I'm still of the opinion that the cancel listener should not be triggered (and just cleared) at the start of a new crafting job. So this loop should be removed.